File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.java
class Solution {
    public int signFunc(int pro){
        if (pro>0){
            return 1;
        }else if (pro<0){
            return -1;

        }else{
            return 0;
        }
    }
    public int arraySign(int[] nums) {
int sign = 1; 
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                return 0; 
            } else if (nums[i] < 0) {
                sign = -sign;  
            }
        }
        return sign;
    }
}

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        pro=1
        for i in range(len(nums)):
            pro=pro*nums[i]

        if pro==0:
            return 0
        elif pro<0:
            return -1
        else:
            return 1